README.md

```diff
@@ -9,6 +9,71 @@ OAuth2 Java Server Library is a backbone of OAuth2 provider which will provide b
 # Supported Flows
  * Client authorization with authorization_code
  * 2-legged oauth2 authorisation with JWT
+ 
+ # Example config servlet for usage of the OAuth2 server library with Guice 
+ ## OAuth2SupportServlet
+ ```Java
+ class OAuth2SupportServlet @Inject constructor(private val tokens: Tokens,
+                                                private val jwtKeyStore: JwtKeyStore,
+                                                private val keyStore: KeyStore,
+                                                private val identityFinder: IdentityFinder,
+                                                private val resourceOwnerIdentityFinder: ResourceOwnerIdentityFinder,
+                                                private val clientFinder: ClientFinder,
+                                                private val clientAuthorizationRepository: ClientAuthorizationRepository) : OAuth2Servlet() {
+   override fun config(): OAuth2Config {
+     return OAuth2Config.newConfig()
+             .tokens(tokens)
+             .jwtKeyStore(jwtKeyStore)
+             .keyStore(keyStore)
+             .identityFinder(identityFinder)
+             .resourceOwnerIdentityFinder(resourceOwnerIdentityFinder)
+             .clientAuthorizationRepository(clientAuthorizationRepository)
+             .clientFinder(clientFinder)
+             .loginPageUrl("/ServiceLogin?continue=")
+             .build()
+   }
+ }
+ 
+ ```
+
+## The responding interfaces are responsible for:
+
+#### IdentityFinder
+Used to find the identity of the client that is requesting authorization.
+
+#### ClientAuthorizationRepository
+Used to generate and persist auth codes for authorized clients and to use them later in the Access token request step.
+
+#### Tokens
+Used to generate Access and Refresh tokens and persist them for later use when a protected resource is requested.
+
+#### JwtKeyStore
+Used to find the Key blocks for verifying JWT authorizations
+
+#### KeyStore
+Used to find the keys used for signing and verifying of the signatures of the id_tokens.
+
+#### ResourceOwnerIdentityFinder
+Used during the authorization of the request to find the identity.    
+
+#### ClientFinder
+Used to persist and find OAuth Clients
+
+## Endpoints in the OAuth2Servlet
```

Comment:
If OAuth2 Servlet is bound to "/o/oauth2/v1/*" then supported endpoints are:

*  /o/oauth2/v1/auth - authorization endpoint
* /o/oauth2/v1/token - token endpoint
* /o/oauth2/v1/userInfo - user info endpoint
* /o/oauth2/v1/tokenInfo - token info endpoint
* /o/oauth2/v1/certs - serves public certs

Reply:
I could just add a description that at the "." is the path to the bound Servlet or i could add this example with the info that the servlet is bound to /o/oauth2/v1/*

Reply:
Example will show better the intention.

Reply:
Changed.